generator/ServiceClientGeneratorLib/Generators/SourceFiles/ServiceConfig.tt

```diff
@@ -135,6 +135,9 @@ namespace <#=this.Config.Namespace#>
                 ClientConfig = this,
                 OriginalRequest = parameters.Request,
                 Request = new DefaultRequest(parameters.Request, ServiceId)
+                {
```

Comment:
Not related to this PR, but what's the use case for `AlternateEndpoint`? I did not know about it and even had to double check it wasn't added in the PR.

Reply:
It is being used in scenarios like RDS and DocDB where they support copying a database from a different region. In this this code path is getting triggered by the changes `PreSignedUrlRequestHandler.cs` classes. We essentially have to create a presigned URL to the resource in the region the database is being copied. So we need use the current client config settings but override the region with the region the database is in.

In V3 the URL for the presigned URL was being generated by the old obsolete method that didn't use EP 2.0.